sql_to_code/parsers/create_table/parser.py

```diff
@@ -1,46 +1,67 @@
-import re
-from typing import List
+from pyparsing import *
 
-from .models import Attribute, Table, default_type
+from .models import Attribute, Table
 
-regex_name_schema: str = r'CREATE TABLE "([\S\d]+)"\s*\(\s*(.+)\s*\);'
-name_and_type_regex = r"(^[\S\d]+)\s([\S]+)\s*"
-default_regex = r"DEFAULT\s+(.?)+\s*"
+# table schema
+create_table = CaselessKeyword("create table")
+table_name = MatchFirst(QuotedString('"'))("table_name")
 
+# field schema
+column_name = QuotedString('"')("column_name")
 
-def parse(sql_text: str) -> Table:
-    table_name, schema = re.findall(regex_name_schema, sql_text)[0]
-    all_attributes: List[str] = schema.split(",")
-    schema: List[str] = map(
-        lambda attribute: attribute.strip().replace('"', ""), all_attributes
-    )
-    attributes = parse_attributes(schema)
+# column type:
+# "(" ")" - because of varchar(40)
+# "_" - because of enums like process_type
+column_type = Word(alphas, alphanums + "(" + ")" + "_")("column_type")
```

Comment:
I think here is problem either with naming (it should be called VARCHAR for example) or value (it should contain all possible column type then). What about name it VARCHAR and in case of `column type` we could create something like:

column_type = OneOf(list(VARCHAR, [REALNUMBER](https://github.com/pyparsing/pyparsing/blob/5713fba4fc952cd08a136301ff84275f19e1e930/pyparsing/common.py#L187), [INTEGER](https://github.com/pyparsing/pyparsing/blob/5713fba4fc952cd08a136301ff84275f19e1e930/pyparsing/common.py#L162), ENUM (create it in the same way as VARCHAR was created ))

Guess this separation by tokens will be easy to read. Also, enum related stuff is already specified in [enumSchema](https://github.com/OriginalUtkin/sql_generator/pull/27/files#diff-9996898c81282bd651d4014d7b0f746633e614b48f6e3651042ce861e41fbfdcR5)

Reply:
Btw they have overridden methods such as `[or](https://github.com/pyparsing/pyparsing/blob/5713fba4fc952cd08a136301ff84275f19e1e930/pyparsing/core.py#L1216)`, `[and](https://github.com/pyparsing/pyparsing/blob/5713fba4fc952cd08a136301ff84275f19e1e930/pyparsing/core.py#L1275)` and some other so we could take an beneficial of it

Reply:
Wow, those guys are even have simple example of parser for SQL
So, we could just make it in similar way as they did [there](https://github.com/pyparsing/pyparsing/blob/5713fba4fc952cd08a136301ff84275f19e1e930/examples/simpleSQL.py#L48)

Reply:
for now we don't have support of parsing results, so we might decide what fields we should support and split this feature into separated pull request